packages/transactions/src/OutboundShipment/DetailView/DetailView.tsx

```diff
@@ -48,7 +54,44 @@ const placeholderTransaction: Transaction = {
   invoiceNumber: '',
 };
 
-const useDraft = createDraftStore<Transaction>();
+const draftCreator = <T extends Transaction>(
+  original: T,
+  setter: SetState<DraftStore<T>>
+) => {
+  const mappedLines = original.items?.map(line => ({
+    ...line,
+    setQuantity: (rowIdx: number, value: number) => {
+      setter((state: DraftStore<Transaction>) => {
+        if (!state.draft?.items) {
+          return state;
+        }
+
+        const newState = {
+          ...state,
+          draft: {
+            ...state.draft,
+            items: [...state.draft.items],
+          },
+        };
+
+        if (newState.draft.items[rowIdx]) {
+          newState.draft.items[rowIdx] = {
+            ...(newState.draft.items[rowIdx] as Item), // TODO: Type here
+            quantity: value,
+          };
+        }
+
+        return newState;
+      });
+    },
+  }));
+
+  const newDraft = { ...original, items: mappedLines };
+
+  return newDraft;
+};
```

Comment:
So, added to `DraftStore` zustand store, the method `createDraft`. This is required when you create the store (so, need to create a method like this for any type of store, transaction, location etc) and is called after we receive a response from a request for data. This function is passed the zustand state setter, and injects each row with the `setQuantity` function.

In general I really like this and think it's nice and simple way to get the updater to a cell and even becomes type safe.

I wonder though if there are better ways to be doing this (MobX, valtio?, plain classes?, MobX State Tree?). I've not seen this approach before and I suspect it's because it seems very similar to a mobx style approach (Where, we wrap the response in some mobx class, and pass that down the tree) but instead of wrapping each cell in an `observe` HOC, we still trigger the normal render

Reply:
could also make this simpler with https://immerjs.github.io/immer/produce maybe.

It's been awhile since I'd done this immutable style updating. I really hate it 🤣